client/db/bolt/db.go

```diff
@@ -220,8 +220,26 @@ func (db *BoltDB) Run(ctx context.Context) {
 		db.log.Errorf("Unable to backup database: %v", err)
 	}
 
-	// Only compact the current DB file if there are excessive free pages.
-	if db.Stats().FreePageN < 32 { // 128 KiB for 4096 page size
+	// Only compact the current DB file if there is excessive free space, in
+	// terms of bytes AND relative to total DB size.
+	const byteThresh = 1 << 18 // 256 KiB free
+	const pctThresh = 0.05     // 5% free
```

Comment:
Note that `pctThresh` kicks in, effectively taking the place of `byteThresh`, at ~5 MB DB size.